### PR TITLE
Remove public key from the ed25519 signing API

### PR DESCRIPTION
**Pull request by andrewkozlik**
Resolves https://github.com/trezor/trezor-firmware/issues/2338.

### CODE REVIEW

File: crypto/ed25519-donna/ed25519.h [hunk 1]
```diff
@@ -21,9 +21,9 @@ void ed25519_publickey_ext(const ed25519_secret_key sk, const ed25519_secret_key
 #endif
 
 int ed25519_sign_open(const unsigned char *m, size_t mlen, const ed25519_public_key pk, const ed25519_signature RS);
```
Comment (kchalkias) [inline crypto/ed25519-donna/ed25519.h:23]: Thanks for the refactoring, shall we also update this one?

Comment (andrewkozlik) [reply crypto/ed25519-donna/ed25519.h:23]: This is the signature verification function, so I think there is nothing to update here. The naming is confusing, I know.

Comment (kchalkias) [reply crypto/ed25519-donna/ed25519.h:23]: Oh I see, what about `ed25519_cosi_sign`?

Comment (andrewkozlik) [reply crypto/ed25519-donna/ed25519.h:23]: CoSi is Cothority Collective Signing, it's an aggregate multisig scheme and the signing function needs the _collective public key_ `pk`, see https://github.com/dedis/doc/blob/master/cosi/cosi.md#signature-generation. So in this case having `pk` in the API is justified.

**Review by kchalkias — APPROVED** [review]
Thanks, great work and blazing fast response!

File: crypto/ed25519-donna/ed25519.h (hunk 1 above)
Comment (kchalkias) [reply crypto/ed25519-donna/ed25519.h:23]: Confirmed that in cosi signature scheme, we expect an extra randomness and the R value is not deterministically generated by M and sk as in the original ed25519. 

That said, assuming that the sign function will be invoked with a fresh tuple of <random nonce, pk> each time we invoke it, then the current implementation is safe. Can you confirm that if we ever retry cosi sign for the same message, the nonce is always a fresh random number?

**Review by prusnak — APPROVED** [review]
LGTM, thanks Andrew

**Pull request merged**